ios/browser/api/networking/models/extensions/BraveCertificateGenericExtensionModel.mm

```diff
@@ -0,0 +1,105 @@
+  }
+  return self;
+}
+@end
+
+@implementation BraveCertificateGenericExtensionModel
+- (void)parseExtension:(X509_EXTENSION*)extension {
+  std::unique_ptr<x509_utils::X509_EXTENSION_INFO> info =
+                                      x509_utils::decode_extension_info(extension);
+  if (info) {
+    if (info->get_type() == x509_utils::X509_EXTENSION_INFO::TYPE::STRING) {
+      std::string* string_ptr = info->get_extension_string();
+      _extensionType = BraveGenericExtensionType_STRING;
+      _stringValue = string_ptr ? brave::string_to_ns(*string_ptr) : [[NSString alloc] init];
+      return;
+    } else if (info->get_type() == x509_utils::X509_EXTENSION_INFO::TYPE::MULTI_VALUE) {
+      _extensionType = BraveGenericExtensionType_KEY_VALUE;
+      
+      NSMutableArray* pairs = [[NSMutableArray alloc] init];
+      std::vector<std::pair<std::string, std::string>>* vec = info->get_extension_multi_value();
+      for (auto it = vec->begin(); it != vec->end(); ++it) {
+        [pairs addObject:[[BraveCertificateGenericExtensionPairModel alloc] initWithKey:
+                                                                                  brave::string_to_ns(it->first)
+                                                                              withValue:brave::string_to_ns(it->second)]];
+      }
+      _arrayValue = pairs;
+      return;
+    }
+  }
+    
+  _extensionType = BraveGenericExtensionType_HEX_STRING;
+  ASN1_OCTET_STRING* data = X509_EXTENSION_get_data(extension);
+  _stringValue = data ? brave::string_to_ns(x509_utils::hex_string_from_ASN1STRING(data)) : [[NSString alloc] init];
+}
+
+- (void)debugPrint:(X509_EXTENSION*)extension {
+  std::string result;
+  
+  std::unique_ptr<x509_utils::X509_EXTENSION_INFO> info =
+                                      x509_utils::decode_extension_info(extension);
+  if (info) {
+    if (info->get_type() == x509_utils::X509_EXTENSION_INFO::TYPE::STRING) {
+      result += *info->get_extension_string();
+      result += "\n";
+    } else if (info->get_type() == x509_utils::X509_EXTENSION_INFO::TYPE::MULTI_VALUE) {
+      std::vector<std::pair<std::string, std::string>>* vec = info->get_extension_multi_value();
+      for (auto it = vec->begin(); it != vec->end(); ++it) {
+        if (it->first.empty()) {
+          result += it->second;
+        } else if (it->second.empty()) {
+          result += it->first;
+        } else {
+          result += it->first + ": ";
+          result += it->second;
+        }
+
+        result += "\n";
+      }
+    } else {
+      ASN1_OCTET_STRING* data = X509_EXTENSION_get_data(extension);
+      if (data) {
+        result += x509_utils::hex_string_from_ASN1STRING(data);
+        result += "\n";
+      }
+    }
+  } else {
+    ASN1_OCTET_STRING* data = X509_EXTENSION_get_data(extension);
+    if (data) {
+      result += x509_utils::hex_string_from_ASN1STRING(data);
+      result += "\n";
+    }
+  }
+
+  printf("%s\n", result.c_str());
```

Comment:
If `stdout` visible anywhere? Just curious- maybe when running in a simulator you can see it? Not familiar enough with iOS 😄

Reply:
It will be visible in Simulator. However, I changed it to `NSLog` which is probably better as it'd be visible always. This is debugging code though, so I also wrapped it in an `if defined(DEBUG)`